components/partials/derivatives/trading/trade.vue

```diff
@@ -1499,10 +1506,34 @@ export default Vue.extend({
       }
 
       this.form.amount = new BigNumberInBase(form.amount || 0).toFixed(
-        market.quantityDecimals
+        market.quantityDecimals,
+        BigNumberInBase.ROUND_DOWN
       )
     },
 
+    onAmountKeydown(event: DOMEvent<HTMLInputElement>) {
+      const { market, form } = this
+
+      if (!market) {
+        return
+      }
+
+      // prevent dot for quantity decimals 0
+      if (market.quantityDecimals === 0 && isDotKeycode(event.keyCode)) {
```

Comment:
Instead of early return here and multiple conditionals that will be hard to understand in the future let's do the conditionals in 2 different constants that explain their meaning (for example `const inputIsADot = ... ` or something like that (use your best judgment) and then just check `conditional1 || conditional2` to do `event.preventDefault()`. Do the same for the spot.